rust/toolchain.bzl

```diff
@@ -268,6 +268,15 @@ def _rust_toolchain_impl(ctx):
         linking_context = linking_context,
     )
 
+    # Calculate the rustc sysroot path by using a file from the rust-std bundle
+    rust_std_files_list = rust_std.files.to_list()
```

Comment:
Can this be empty? If so, let's add a nice `fail` error message?

Reply:
I don't know if it'd be valid for this to be empty but I do think it's valid for `rust_std` to be `None` in the case of `no_std` builds when building for more obscure targets. I can update the code to express that if it makes sense to you too.

Reply:
I ended up adding the changes to the last commit but the question still stands and can update accordingly

Reply:
Some extra context. To my knowledge the sysroot is only useful for locating `rust_std`. `rustc` should otherwise be able to function the same without it being explicitly set.